src/functional/program_processors/codegens/gtfn/gtfn_ir_to_gtfn_im_ir.py

```diff
@@ -0,0 +1,274 @@
+            lam_idx = self.uids.sequential_id(prefix="lam")
+            params = [self.visit(param, **kwargs) for param in node.fun.params]
+            args = [self.visit(arg, **kwargs) for arg in node.args]
+            for param, arg in zip(params, args):
+                if param.id in self.sym_table:
+                    self.localized_symbols.append(param.id)
+                self.imp_list_ir.append(
+                    InitStmt(
+                        lhs=gtfn_ir_common.Sym(id=f"{param.id}_local")
+                        if param.id in self.sym_table
+                        else gtfn_ir_common.Sym(id=f"{param.id}"),
+                        rhs=arg,
+                    )
+                )
+            expr = self.visit(node.fun.expr, **kwargs)
+            self.imp_list_ir.append(InitStmt(lhs=gtfn_ir_common.Sym(id=f"{lam_idx}"), rhs=expr))
+            return gtfn_ir_common.SymRef(id=f"{lam_idx}")
+        if _is_reduce(node):
+            return self.handle_Reduction(node, **kwargs)
+        if (
+            isinstance(node.fun, gtfn_ir_common.SymRef) and node.fun.id == "make_tuple"
+        ):  # TODO: bad hardcoded string
+            tupl_idx = self.uids.sequential_id(prefix="tupl")
+            for i, arg in enumerate(node.args):
+                expr = self.visit(arg, **kwargs)
+                self.imp_list_ir.append(
+                    InitStmt(lhs=gtfn_ir_common.Sym(id=f"{tupl_idx}_{i}"), rhs=expr)
+                )
+            tup_args = [gtfn_ir_common.SymRef(id=f"{tupl_idx}_{i}") for i in range(len(node.args))]
+            tuple_fun = gtfn_ir.FunCall(fun=gtfn_ir_common.SymRef(id="make_tuple"), args=tup_args)
+            self.imp_list_ir.append(
+                InitStmt(lhs=gtfn_ir_common.Sym(id=f"{tupl_idx}"), rhs=tuple_fun)
+            )
+            return gtfn_ir_common.SymRef(id=f"{tupl_idx}")
+        return gtfn_ir.FunCall(
+            fun=self.visit(node.fun, **kwargs),
+            args=[self.visit(arg, **kwargs) for arg in node.args],
+        )
+
+    def visit_TernaryExpr(self, node: gtfn_ir.TernaryExpr, **kwargs):
+        cond = self.visit(node.cond, **kwargs)
+        if_ = self.visit(node.true_expr, **kwargs)
+        else_ = self.visit(node.false_expr, **kwargs)
+        cond_idx = self.uids.sequential_id(prefix="cond")
+        self.imp_list_ir.append(
+            Conditional(
+                cond_type=f"{cond_idx}_t",
+                init_stmt=InitStmt(
+                    init_type=f"{cond_idx}_t",
+                    lhs=gtfn_ir_common.Sym(id=cond_idx),
+                    rhs=gtfn_ir.Literal(value="0.", type="float64"),
+                ),
+                cond=cond,
+                if_stmt=AssignStmt(lhs=gtfn_ir_common.SymRef(id=cond_idx), rhs=if_),
+                else_stmt=AssignStmt(lhs=gtfn_ir_common.SymRef(id=cond_idx), rhs=else_),
+            )
+        )
+        return gtfn_ir_common.SymRef(id=cond_idx)
+
+
+@dataclasses.dataclass(frozen=True)
```

Comment:
Yeah, monkey see -> monkey do effect. Every Visitor I encountered while perusing the code base at that point in time was a dataclass, so I made mine a dataclass too because I thought its canonical. I now notice that this is not at all the case and there are plenty visitors (and translators) not marked as dataclass